youtube_dl/extractor/loom.py

```diff
@@ -0,0 +1,202 @@
+                'upload_date': '20200423',
+                'timestamp': 1587646164
+            }
+        },
+        {
+            'url': 'https://www.loom.com/share/6670e3eba3c84dc09ada8306c7138075',
+            'md5': 'bfad8181ed49d6252b10dfdeb46c535e',
+            'info_dict': {
+                'id': '6670e3eba3c84dc09ada8306c7138075',
+                'ext': 'mp4',
+                'title': 'How to record your first video on Loom',
+                'uploader': 'Allie Hitchcock',
+                'upload_date': '20201118',
+                'timestamp': 1605729404
+            }
+        }
+    ]
+
+    def _extract_video_info_json(self, webpage, video_id):
+        info = self._html_search_regex(
+            r'window.loomSSRVideo = (.+?);',
+            webpage,
+            'info')
+        return self._parse_json(info, 'json', js_to_json)
+
+    def _get_url_by_id_type(self, video_id, type):
+        request = compat_urllib_request.Request(
+            self._BASE_URL + 'api/campaigns/sessions/' + video_id + '/' + type,
+            {})
+        json_doc = self._download_json(request, video_id)
+        return (url_or_none(json_doc.get('url')), json_doc.get('part_credentials'))
+
+    def _get_m3u8_formats(self, url, video_id, credentials):
+        format_list = self._extract_m3u8_formats(url, video_id)
+        for item in format_list:
+            item['protocol'] = 'm3u8_native'
+            item['url'] += '?' + credentials
+            item['ext'] = 'mp4'
+            item['format_id'] = 'hls-' + str(item.get('height', 0))
+            item['extra_param_to_segment_url'] = credentials
+        return format_list
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        info = self._extract_video_info_json(webpage, video_id)
+
+        formats = []
+        for type in ['transcoded-url', 'raw-url']:
+            (url, part_credentials) = self._get_url_by_id_type(video_id, type)
+            ext = self._search_regex(
+                r'\.([a-zA-Z0-9]+)\?',
+                url, 'ext', default=None)
+            if(ext != 'm3u8'):
+                formats.append({
+                    'url': url,
+                    'ext': ext,
+                    'format_id': type,
+                    'width': try_get(info, lambda x: x['video_properties']['width']),
+                    'height': try_get(info, lambda x: x['video_properties']['height'])
+                })
+            else:
+                credentials = compat_urllib_parse_urlencode(part_credentials)
+                m3u8_formats = self._get_m3u8_formats(url, video_id, credentials)
+                for i in range(len(m3u8_formats)):
+                    formats.insert(
+                        (-1, len(formats))[i == len(m3u8_formats) - 1],
+                        m3u8_formats[i])
+
+        return {
+            'id': info.get('id'),
+            'title': info.get('name'),
```

Comment:
Mandatory. Read coding conventions.

Reply:
For the `id`, I may provide a fallback value from the url. However, the `title` does not have another fallback source, other than the embeded JSON.

Any advice?